### DISCUSSION TITLE
Pytorch Lightning doesn't have CUDA?

### DISCUSSION THREAD

**Discussion by Troceleng** [Lightning-AI/pytorch-lightning#7158] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
## 🐛 Bug

Hello, I'm trying to use Pytorch Lightning in order to speed up my ESR GAN renders on Windows 10. 

However, when I ran the installation code and attempt to run Cupscale (which I use as a GUI for ESR GAN), I get an error saying "Pytorch compiled without CUDA".

Is there a way to choose to install specifically the CUDA version of Pytorch with Lightning install, or are the two incompatible? If the latter is the case, that's not good to hear. If the former, can I have such a code?

**Comment by Borda** [comment, marked as answer]
@TrocelengStudios ho this seems like you do not have CUDA installed, can you run `nvidia-smi`?
cc: @awaelchli

**Reply by Troceleng** [reply]
I get this:
NVIDIA-SMI 462.31       Driver Version: 462.31       CUDA Version: 11.2

What should I get?

**Reply by awaelchli** [reply]
Go to the pytorch website and select the installation with cuda. It's directly on the homepage pytorch.org

**Reply by Troceleng** [reply]
Alright, I installed it from there. Problem is that someone told me that Pytorch Lightning is faster and isn't this now just the official Pytorch?

**Reply by awaelchli** [reply]
Lightning is built on top of PyTorch, so it requires a pytorch installation :) Lightning is not by itself faster than PyTorch, it can't really be. But it can accelerate your workflow because generally you need to write less code and you stay organized :) 

I suggest you familiarize yourself first with pytorch by following some tutorials: https://pytorch.org/tutorials/
Then once you have implemented your first pytorch model, you can try Lightning and you will see the many benefits.

**Reply by Troceleng** [reply]
Okay, it's too bad it won't make my pre-existing Pytorch-accelerated stuff run faster, but I guess it won't hurt having it installed in case I want to make some code myself as well or train models for ESR GAN (my main Pytorch application). Thank you for the advice and I will look into those tutorials. :) Have a nice spring.
